x-pack/plugins/spaces/server/capabilities/capabilities_switcher.ts

```diff
@@ -16,9 +16,7 @@ export function setupCapabilitiesSwitcher(
   logger: Logger
 ): CapabilitiesSwitcher {
   return async (request, capabilities) => {
-    const isAnonymousRequest = !request.route.options.authRequired;
-
-    if (isAnonymousRequest) {
+    if (!request.auth.isAuthenticated) {
```

Comment:
We may have to check if security is enabled before bailing here. Luckily (?), Spaces declares an optional dependency on security, so we can check if security is enabled, _and_ if it is an authenticated request

Reply:
@legrego our test verifies another behavior.
https://github.com/elastic/kibana/blob/d66b3c74eb18d72208d0652a0b5f72cd9701191b/x-pack/test/ui_capabilities/spaces_only/tests/catalogue.ts#L25
It doesn't rely on security plugin availability.

Reply:
Our goal here is to skip toggling capabilities if we don't have credentials, but would otherwise need them in order to perform the capabilities check. We used to silently fail in the past, but we made this more explicit in https://github.com/elastic/kibana/pull/57693#discussion_r379575784.

So we need to figure out a way to determine if credentials are required in order to toggle capabilities. Prior to https://github.com/elastic/kibana/pull/58589, we had two distinct routes to retrieve capabilities: one which required auth, and another which didn't. This made the check easy for us here, but it's no longer sufficient now that we have a single route with `optional` auth.

When security is disabled, then I believe `request.auth.isAuthenticated` will always be `false`, but in this case, we will still want to attempt to toggle capabilities.

When security is enabled, then we only want to attempt to toggle capabilities when `request.auth.isAuthenticated` is `true`

Reply:
> When security is disabled, then I believe request.auth.isAuthenticated will always be false, but in this case, we will still want to attempt to toggle capabilities.

It sounds like you want to know [AuthState](https://github.com/elastic/kibana/blob/e00f2628af71cd3d39e04dd56a3509b86d51af36/src/core/server/http/auth_state_storage.ts#L25), which is available via [auth api](https://github.com/elastic/kibana/blob/5c436534d61e20c88fbecb1108e674c8d413d381/src/core/server/http/http_server.ts#L57)
And you need to retrieve capabilities for `authenticated` & `unknown` statuses. Is it right? Moreover, Spaces plugin will log the same warning. Then the current PR can wait or we even can keep the current behavior from the master.

Reply:
> > When security is disabled, then I believe request.auth.isAuthenticated will always be false, but in this case, we will still want to attempt to toggle capabilities.
> 
> It sounds like you want to know [AuthState](https://github.com/elastic/kibana/blob/e00f2628af71cd3d39e04dd56a3509b86d51af36/src/core/server/http/auth_state_storage.ts#L25), which is available via [auth api](https://github.com/elastic/kibana/blob/5c436534d61e20c88fbecb1108e674c8d413d381/src/core/server/http/http_server.ts#L57)
> And you need to retrieve capabilities for `authenticated` & `unknown` statuses. Is it right? Moreover, Spaces plugin will log the same warning. 

Yes, that sounds correct to me

> Then the current PR can wait or we even can keep the current behavior from the master.

Perhaps it'd be worthwhile to reduce this message from `warn` to `debug` then, if this is a message we're going to expect in certain conditions. I don't want to pollute the logs unnecessarily.